analyzers/src/SonarAnalyzer.Common/Extensions/RegexExtensions.cs

```diff
@@ -42,23 +44,59 @@ public static Match SafeMatch(this Regex regex, string input)
     /// <summary>
     /// Matches the input to the regex. Returns <see langword="false" /> in case of an <see cref="RegexMatchTimeoutException" />.
     /// </summary>
-    public static bool SafeIsMatch(this Regex regex, string input)
+    public static bool SafeIsMatch(this Regex regex, string input) =>
+        regex.SafeIsMatch(input, false);
+
+    /// <summary>
+    /// Matches the input to the regex. Returns <paramref name="timeoutFallback"/> in case of an <see cref="RegexMatchTimeoutException" />.
+    /// </summary>
+    public static bool SafeIsMatch(this Regex regex, string input, bool timeoutFallback)
     {
         try
         {
             return regex.IsMatch(input);
         }
         catch (RegexMatchTimeoutException)
         {
-            return false;
+            return timeoutFallback;
         }
     }
 
-    public static bool SafeIsMatch(string input, string pattern, TimeSpan timeout, RegexOptions options = RegexOptions.None)
+    /// <summary>
+    /// Matches the input to the regex. Returns an empty <see cref="MatchCollection" /> in case of an <see cref="RegexMatchTimeoutException" />.
+    /// </summary>
+    public static MatchCollection SafeMatches(this Regex regex, string input)
+    {
+        try
+        {
+            _ = regex.Matches(input).Count; // MatchCollection is lazy. Accessing "Count" executes the regex and caches the result
+            return regex.Matches(input);
```

Comment:
I did not know this!

Shouldn't it be written as this though?
```csharp
var res = regex.Matches(input);
_ = res.Count; // MatchCollection is lazy. Accessing "Count" executes the regex and caches the result
return res;
```
Just to avoid one extra allocation, by referring to the same match collection?